Sources/Zip/QuickZip.swift

```diff
@@ -49,11 +40,8 @@ extension Zip {
      - Returns: `URL` of the destination folder.
      */
     public class func quickUnzipFile(_ path: URL, progress: ((_ progress: Double) -> ())?) throws -> URL {
-        let fileExtension = path.pathExtension
-        let fileName = path.lastPathComponent
-        let directoryName = fileName.replacingOccurrences(of: ".\(fileExtension)", with: "")
-        let documentsUrl = FileManager.default.urls(for: self.searchPathDirectory, in: .userDomainMask)[0]
-        let destinationUrl = documentsUrl.appendingPathComponent(directoryName, isDirectory: true)
+        let directoryName = path.lastPathComponent.replacingOccurrences(of: ".\(path.pathExtension)", with: "")
```

Comment:
This `replacingOccurrences(of:with:)` could result in weird bugs when the name also contains the extension for some reason.
I'd rather explicitly remove only the known extension (with `dropLast()` perhaps).
Will also be more performant.

Reply:
Thanks Mahdi! Seems like the best way to achieve this is with `path.deletingPathExtension().lastPathComponent`.

Your comment encouraged me to add similar controls to the `quickZip` function as well